prow/scripts/cluster-integration/kyma-gke-end-to-end-test.sh

```diff
@@ -0,0 +1,247 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o pipefail  # Fail a pipe if any sub-command fails.
+
+discoverUnsetVar=false
+
+for var in INPUT_CLUSTER_NAME REPO_OWNER_GIT REPO_NAME_GIT DOCKER_PUSH_REPOSITORY KYMA_PROJECT_DIR CLOUDSDK_CORE_PROJECT CLOUDSDK_COMPUTE_REGION CLOUDSDK_DNS_ZONE_NAME GOOGLE_APPLICATION_CREDENTIALS KYMA_ARTIFACTS_BUCKET KYMA_BACKUP_RESTORE_BUCKET KYMA_BACKUP_CREDENTIALS CLOUDSDK_COMPUTE_ZONE; do
+    if [ -z "${!var}" ] ; then
+        echo "ERROR: $var is not set"
+        discoverUnsetVar=true
+    fi
+done
+if [ "${discoverUnsetVar}" = true ] ; then
+    exit 1
+fi
+
+export TEST_INFRA_SOURCES_DIR="${KYMA_PROJECT_DIR}/test-infra"
+export TEST_INFRA_CLUSTER_INTEGRATION_SCRIPTS="${TEST_INFRA_SOURCES_DIR}/prow/scripts/cluster-integration/helpers"
+export KYMA_SOURCES_DIR="${KYMA_PROJECT_DIR}/kyma"
+export KYMA_SCRIPTS_DIR="${KYMA_SOURCES_DIR}/installation/scripts"
+
+### For provision-gke-cluster.sh
+export GCLOUD_PROJECT_NAME="${CLOUDSDK_CORE_PROJECT}"
+export GCLOUD_COMPUTE_ZONE="${CLOUDSDK_COMPUTE_ZONE}"
+export GCLOUD_SERVICE_KEY_PATH="${GOOGLE_APPLICATION_CREDENTIALS}"
+
+### For generate-cluster-backup-config.sh
+export BACKUP_CREDENTIALS="${KYMA_BACKUP_CREDENTIALS}"
+export BACKUP_RESTORE_BUCKET="${KYMA_BACKUP_RESTORE_BUCKET}"
+
+readonly REPO_OWNER=$(echo "${REPO_OWNER_GIT}" | tr "[:upper:]" "[:lower:]")
+readonly REPO_NAME_GIT=$(echo "${REPO_NAME_GIT}" | tr "[:upper:]" "[:lower:]")
+readonly CURRENT_TIMESTAMP=$(date +%Y%m%d)
+
+readonly STANDARIZED_NAME=$(echo "${INPUT_CLUSTER_NAME}" | tr "[:upper:]" "[:lower:]")
+readonly DNS_SUBDOMAIN="${STANDARIZED_NAME}"
+export CLUSTER_NAME="${STANDARIZED_NAME}"
+
+# shellcheck disable=SC1090
+source "${TEST_INFRA_SOURCES_DIR}/prow/scripts/library.sh"
+
+removeCluster() {
+    #Turn off exit-on-error so that next step is executed even if previous one fails.
+    set +e
+
+    EXIT_STATUS=$?
```

Comment:
removeCluster method shouldn't return exit status, especially it should never fail. please remove